third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -193,6 +195,45 @@ ComputationClient::DataPtr PjRtComputationClient::CopyToDevice(
                                     std::move(status_or.value()));
 }
 
+ComputationClient::DataPtr PjRtComputationClient::ReplicateShardedData(
```

Comment:
If I understand correctly,  this functions takes a sharded data and then replicate it on all device and then return? This is useful when you want to all_gather a sharded data and return to the user?

Reply:
Correct, basically we perform a simple/light identity calculation, so that the compiler replicates the output (reassembled input).